specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/virtualWan.json

```diff
@@ -2634,6 +2634,231 @@
+          },
+          {
+            "name": "connectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn connection."
+          },
+          {
+            "name": "linkConnectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn link connection."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. Returns the shared key of VpnLink connection.",
+            "schema": {
+              "$ref": "#/definitions/ConnectionSharedKey"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "VpnSiteLinkConnectionSharedKeyGet": {
+            "$ref": "./examples/VpnSiteLinkConnectionSharedKeyGet.json"
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "vpnLinkConnections"
+        ],
+        "operationId": "VpnLinkConnections_Put_SharedKey",
+        "description": "Put the shared key of VpnLink connection specified.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "gatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the gateway."
+          },
+          {
+            "name": "connectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn connection."
+          },
+          {
+            "name": "linkConnectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn link connection."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConnectionSharedKey"
+            },
+            "description": "Parameters supplied to the Begin Set vpn site link connection Shared key operation throughNetwork resource provider."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
```

Comment:
>responses [](http://example.com/codeflow?start=9&length=9)

201 is not a valid POST response. Please follow the RPC for async POST. Ref: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#call-action-post-asynchronously). Also, POST async response can't have a payload.

Reply:
Thanks for sharing this. I fixed it